AIAgent/ml/training/validation.py

```diff
@@ -82,33 +82,17 @@ def validate_coverage(
             if isinstance(result, GameError):
                 need_to_save_map: bool = result.need_to_save_map()
                 if not need_to_save_map:
-                    statistics_collector.fail(result._map)
-            else:
-                all_results.append(result)
-    statistics_collector.update_results(all_results)
+                    maps_to_remove.append(result.map2result.map)
```

Comment:
`maps_to_remove` will be used to check whether should we fail immediately but here we use additional flag `need_to_save_map` to fill this collection. Looks strange.